rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/model/URITemplate.java

```diff
@@ -130,10 +130,31 @@ public String getPatternValue() {
 
     private static String escapeCharacters(String expression) {
 
-        StringBuilder sb = new StringBuilder();
-        for (int i = 0; i < expression.length(); i++) {
-            char ch = expression.charAt(i);
-            sb.append(isReservedCharacter(ch) ? "\\" + ch : ch);
+        int length = expression.length();
+        int i = 0;
+        char ch = ' ';
+        for (; i < length; ++i) {
+            ch = expression.charAt(i);
+            if (isReservedCharacter(ch)) {
+                break;
+            }
+        }
+        
+        if (i == length) {
+            return expression;
+        }
+        
+        StringBuilder sb = new StringBuilder(length + 8);
```

Comment:
Why `length + 8`?

Reply:
There should be a comment in the code in that case explaining why "8" is chosen